lib/plenario_aot/aot_actions.ex

```diff
@@ -0,0 +1,153 @@
+
+  alias PlenarioAot.{AotMeta, AotData, AotObservation}
+
+  def create_meta(network_name, source_url) do
+    params = %{network_name: network_name, source_url: source_url}
+    meta = %AotMeta{}
+
+    Logger.info("Creating new AoT Meta with params `#{inspect(params)}`")
+
+    AotMeta.changeset(meta, params)
+    |> Repo.insert()
+  end
+
+  def list_metas, do: Repo.all(AotMeta)
+
+  def get_meta(identifier) when is_integer(identifier), do: Repo.get_by(AotMeta, id: identifier)
+  def get_meta(identifier) when is_bitstring(identifier), do: Repo.get_by(AotMeta, slug: identifier)
+
+  def update_meta(%AotMeta{} = meta, params \\ []) do
+    params = Enum.into(params, %{})
+
+    Logger.info("Updating AoT Meta `#{meta.network_name}` with #{inspect(params)}")
+
+    AotMeta.changeset(meta, params)
+    |> Repo.update()
+  end
+
+  def compute_and_update_meta_bbox(%AotMeta{} = meta) do
+    query =
+      from d in AotData,
+      where: d.aot_meta_id == ^meta.id,
+      select: %{
+        min_lat: min(d.latitude),
+        max_lat: max(d.latitude),
+        min_lon: min(d.longitude),
+        max_lon: max(d.longitude)
+      }
+    res = Repo.one(query)
+
+    case res do
+      %{min_lat: min_lat, max_lat: max_lat, min_lon: min_lon, max_lon: max_lon} ->
+        bbox = %Geo.Polygon{
+          srid: 4326,
+          coordinates: [[
+            {max_lat, min_lon},
+            {min_lat, min_lon},
+            {min_lat, max_lon},
+            {max_lat, max_lon},
+            {max_lat, min_lon}
+          ]]
+        }
+
+        case update_meta(meta, bbox: bbox) do
+          {:ok, _} ->
+            :ok
+
+          {:error, cs} ->
+            Logger.error("#{inspect(cs)}")
+            :ok
+        end
+
+      _ ->
+        {:error, "No available min/max tuples"}
+    end
+  end
+
+  def compute_and_update_meta_time_range(%AotMeta{} = meta) do
+    query =
+      from d in AotData,
+      where: d.aot_meta_id == ^meta.id,
+      select: %{
+        min_ts: min(d.timestamp),
+        max_ts: max(d.timestamp)
+      }
+    res = Repo.one(query)
+
+    case res do
+      %{min_ts: min_ts, max_ts: max_ts} ->
+        case update_meta(meta, time_range: [min_ts, max_ts]) do
+          {:ok, _} ->
+            :ok
+
+          {:error, cs} ->
+            Logger.error("#{inspect(cs)}")
+            :ok
+        end
+
+      _ ->
+        {:error, "No available timestamp tuple"}
+    end
+  end
+
+  def format_time_range(%AotMeta{time_range: [lower, upper]}) do
+    lower = Timex.format!(lower, "{ISOdate} {ISOtime}")
+    upper = Timex.format!(upper, "{ISOdate} {ISOtime}")
+    "#{lower} to #{upper}"
+  end
+  def format_time_range(_), do: "-"
+
+  def insert_data(%AotMeta{} = meta, %{} = json_payload), do: insert_data(meta.id, json_payload)
+  def insert_data(meta, json_payload) do
+    params = Map.merge(json_payload, %{"aot_meta_id" => meta})
+
+    result =
+      AotData.changeset(params)
+      |> Repo.insert(on_conflict: :nothing)
```

Comment:
Since 2/3 of the data in the payload is playback, we can just skip it. Upserting here takes way too long and there's never going to be a correction to a past observation.